### DISCUSSION TITLE
HexString Byte Converter

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/runtime#55267] [Q&A]
I'd like to make a Json converter that will convert between a hex string and a collection of bytes.

How would I do this in a way that allows me to support byte[], List<byte>, ImmutableArray<byte>, etc without needing to specialize for each different type of collection?

**Comment by Joe4evr** [comment]
You can't construct those types in the same way, so your only option is to specialize.

**Comment by huoyaoyuan** [comment]
Try `JsonConverterFactory`?
